dotnet/src/SemanticKernel.Abstractions/Functions/KernelFunction.cs

```diff
@@ -517,44 +518,24 @@ public AIFunction AsAIFunction(Kernel? kernel = null)
     /// <summary>An <see cref="AIFunction"/> wrapper around a <see cref="KernelFunction"/>.</summary>
     private sealed class KernelAIFunction : AIFunction
     {
+        private static readonly JsonElement s_defaultSchema = JsonDocument.Parse("{}").RootElement;
         private readonly KernelFunction _kernelFunction;
         private readonly Kernel? _kernel;
 
         public KernelAIFunction(KernelFunction kernelFunction, Kernel? kernel)
         {
             this._kernelFunction = kernelFunction;
             this._kernel = kernel;
-
-            string name = string.IsNullOrWhiteSpace(kernelFunction.PluginName) ?
+            this.Name = string.IsNullOrWhiteSpace(kernelFunction.PluginName) ?
                 kernelFunction.Name :
                 $"{kernelFunction.PluginName}-{kernelFunction.Name}";
 
-            this.Metadata = new AIFunctionMetadata(name)
-            {
-                Description = kernelFunction.Description,
-
-                JsonSerializerOptions = kernelFunction.JsonSerializerOptions,
-
-                Parameters = kernelFunction.Metadata.Parameters.Select(p => new AIFunctionParameterMetadata(p.Name)
-                {
-                    Description = p.Description,
-                    ParameterType = p.ParameterType,
-                    IsRequired = p.IsRequired,
-                    HasDefaultValue = p.DefaultValue is not null,
-                    DefaultValue = p.DefaultValue,
-                    Schema = p.Schema?.RootElement,
-                }).ToList(),
-
-                ReturnParameter = new AIFunctionReturnParameterMetadata()
-                {
-                    Description = kernelFunction.Metadata.ReturnParameter.Description,
-                    ParameterType = kernelFunction.Metadata.ReturnParameter.ParameterType,
-                    Schema = kernelFunction.Metadata.ReturnParameter.Schema?.RootElement,
-                },
-            };
+            this.JsonSchema = BuildFunctionSchema(kernelFunction);
```

Comment:
Can this not use the KF's schema if it has one?

Reply:
Sorry I don't follow? I think the only schemas are defined on the individual parameter level.

Reply:
Ah, right